### PR TITLE
fix: updates to optimize tree shaking

### PR DESCRIPTION
**Pull request by joshhowenstine**
## Description

This PR eliminates all re-exports from the codebase and replaces them with direct file references. The primary goal of this change is to improve our project's compatibility with tree shaking. By directly referring to files, we can ensure that only the code actually used in the project is included in the final bundle, which will lead to smaller bundle sizes and better performance in production environments.

## References

NO-JIRA

## Testing

Testing if tree shaking is working properly takes a few steps. You will need to utilize an existing or new project to see the results.

1. Ensure Correct Webpack Configuration: Check that your Webpack configuration is optimized for tree shaking. This includes:

- Utilizing ES6 module syntax (import and export) throughout your project.
- Setting mode to 'production' in your Webpack configuration to enable optimizations like tree shaking.
- Confirming that the sideEffects property in your package.json is properly configured to false or accurately lists fles/modules that do have side effects, aiding Webpack in excluding unused modules.

2. Build the Project: Execute the Webpack build process using npm run build or your project's equivalent script to generate the production bundle. 

3. Review Output: After the build, review the output to ensure there are no errors or warnings related to the tree shaking process or the direct file references. This step is crucial to verify that the modifications have not introduced any build issues. The final bundle should only include Lightning UI components that have been used in your project. All others should no longer be present in the final code.

## Automation

All automation tests should pass as expected. This should have no effect on exported components or functions.

## Checklist

- [x] all commented code has been removed
- [x] any new console issues have been resolved
- [x] code linter and formatter has been run
- [x] test coverage meets repo requirements
- [x] PR name matches the expected semantic-commit syntax

### CODE REVIEW

**Comment by ImCoolNowRight** [comment]
I've built this, brought it into inner source's `develop` branch, built that, and then imported into AirUI. When I built the app, I'm still seeing components that certainly aren't being used showing up in the final bundle. We can pair on this when you're back in the office, but I think either I missed something in testing, or something in the PR needs adjusting.

File: packages/@lightningjs/ui-components/package.json [hunk 1]
```diff
@@ -27,12 +27,14 @@
     "test:watch": "NODE_OPTIONS=--experimental-vm-modules jest --no-coverage --watch --colors"
   },
   "peerDependencies": {
-    "@lightningjs/core": "2.11.0"
+    "@lightningjs/core": "^2.7.0"
```
Comment (ImCoolNowRight) [inline packages/@lightningjs/ui-components/package.json]: What does this need to be rolled back for? It seems like it's still resolving to the same version as before, and I think we needed to keep this locked at 2.11.0 until we looked more into the issues related to #400.

File: packages/@lightningjs/ui-components/src/globals/context/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,33 @@
+type ThemeObject = any;
```
Comment (ImCoolNowRight) [inline packages/@lightningjs/ui-components/src/globals/context/index.d.ts]: Is this ok to be `any`, or should it at least be an `object`?

File: packages/@lightningjs/ui-components/src/mixins/withThemeStyles/utils.js [hunk 3]
```diff
@@ -354,10 +354,10 @@ export const generateSolution = (
   const solution = {};
 
   const uniqueModes = getUniqueProperties([
+    'unfocused', // Unfocused must be at the end for proper fallback since base === 'unfocused' in many cases
```
Comment (ImCoolNowRight) [inline packages/@lightningjs/ui-components/src/mixins/withThemeStyles/utils.js]: Is this comment no longer true?

File: packages/@lightningjs/ui-components/src/textures/index-remove.js
Comment (ImCoolNowRight) [inline packages/@lightningjs/ui-components/src/textures/index-remove.js:1]: Is this file supposed to be removed now?

**Review by ImCoolNowRight — COMMENTED** [review]
Super minor points of clarification, but functionally, I think everything's good to go with this!

File: packages/@lightningjs/ui-components/package.json (hunk 1 above)
Comment (joshhowenstine) [reply packages/@lightningjs/ui-components/package.json]: Will update, thanks!

Comment (joshhowenstine) [reply packages/@lightningjs/ui-components/package.json]: What if we did something like this?

`"@lightningjs/core": ">=2.0.0 <=2.11.0"`

https://docs.npmjs.com/cli/v10/commands/npm-install

That way the consuming application does not have to be locked to a specific version but does not hit the version that has breaking changes.

File: packages/@lightningjs/ui-components/src/globals/context/index.d.ts (hunk 2 above)
Comment (joshhowenstine) [reply packages/@lightningjs/ui-components/src/globals/context/index.d.ts]: Definitely should be object. Thanks

File: packages/@lightningjs/ui-components/src/mixins/withThemeStyles/utils.js (hunk 3 above)
Comment (joshhowenstine) [reply packages/@lightningjs/ui-components/src/mixins/withThemeStyles/utils.js]: Accidental, should not have been committed. Reverted.

File: packages/@lightningjs/ui-components/src/textures/index-remove.js
Comment (joshhowenstine) [reply packages/@lightningjs/ui-components/src/textures/index-remove.js:1]: Had multiple files renamed like this that I failed to remove. They should all be cleaned up now.

**Review by ImCoolNowRight — APPROVED** [review]
Everything looks good!

**Review by slhay28 — APPROVED** [review]

**Pull request merged**